evm/src/main/java/org/hyperledger/besu/evm/precompile/KZGPointEvalPrecompiledContract.java

```diff
@@ -0,0 +1,154 @@
+/*
+ * Copyright Hyperledger Besu Contributors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+package org.hyperledger.besu.evm.precompile;
+
+import static java.nio.file.StandardCopyOption.REPLACE_EXISTING;
+
+import org.hyperledger.besu.evm.frame.ExceptionalHaltReason;
+import org.hyperledger.besu.evm.frame.MessageFrame;
+
+import java.io.BufferedReader;
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import java.io.InputStream;
+import java.nio.charset.Charset;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.util.Optional;
+
+import com.google.common.annotations.VisibleForTesting;
+import ethereum.ckzg4844.CKZG4844JNI;
+import org.apache.tuweni.bytes.Bytes;
+import org.jetbrains.annotations.NotNull;
+
+public class KZGPointEvalPrecompiledContract implements PrecompiledContract {
+
+  public KZGPointEvalPrecompiledContract(final Optional<String> pathToTrustedSetup) {
+
+    String pathToSetup;
+    CKZG4844JNI.Preset bitLength;
+    if (pathToTrustedSetup.isPresent()) {
+      pathToSetup = pathToTrustedSetup.get();
+    } else {
+      InputStream is =
+          KZGPointEvalPrecompiledContract.class.getResourceAsStream(
+              "mainnet_kzg_trusted_setup_4096.txt");
+      try {
+        File jniWillLoadFrom = File.createTempFile("kzgTrustedSetup", "txt");
```

Comment:
you can set delete on exit on this temp file, so it is automatically cleaned

Reply:
I thought createTempFile did that automatically, thanks.